### ISSUE TITLE
year_limited decorator erroneously deletes duplicate data points

### ISSUE THREAD

**Issue by tvwerkhoven** [EnergieID/entsoe-py#161]
When using `EntsoePandasClient.query_day_ahead_prices()` the last item is dropped for some countries, which is properly returned by the `EntsoeRawClient`. Any ideas why this might be?

This works as expected:
```python
# Get today + tomorrow's price
start = pd.Timestamp.today().normalize()
end = start+pd.Timedelta(days=2) # Ask all future data from entsoe
country_code = 'BE'

client = EntsoePandasClient(api_key=secrets['entsoe_token'])
daprice = client.query_day_ahead_prices(country_code, start=start, end=end)
client2 = EntsoeRawClient(api_key=secrets['entsoe_token'])
daprice2 = client2.query_day_ahead_prices(country_code, start=start, end=end)
```

yields the following last two values (correct):
```
[...]
2022-02-23 22:00:00+01:00    140.00
2022-02-23 23:00:00+01:00    129.00
```
and
```xml
[...]
<Point>\n\t\t\t\t\t\t<position>23</position>\n\t\t\t\t\t\t<price.amount>140.00</price.amount>\n\t\t\t\t\t</Point>\n\t\t\t\t\t
<Point>\n\t\t\t\t\t\t<position>24</position>\n\t\t\t\t\t\t<price.amount>129.00</price.amount>\n\t\t\t\t\t</Point>\n\t\t\t
</Period>\n\t</TimeSeries>\n</Publication_MarketDocument>'
```

However if I set `country_code = 'NL'` gives the following last two values (23:00 missing):
```
[...]
2022-02-23 21:00:00+01:00    164.90
2022-02-23 22:00:00+01:00    189.94
```
and (23:00 price present)
```xml
<Point>\n\t\t\t\t\t\t<position>23</position>\n\t\t\t\t\t\t<price.amount>189.94</price.amount>\n\t\t\t\t\t</Point>\n\t\t\t\t\t
<Point>\n\t\t\t\t\t\t<position>24</position>\n\t\t\t\t\t\t<price.amount>140.00</price.amount>\n\t\t\t\t\t</Point>\n\t\t\t
</Period>\n\t</TimeSeries>\n</Publication_MarketDocument>
```

diff of the full output (`diff outNL.xml outDE.xml`) does not give big differences, except for the `L` and `2` suffixes n the `in_Domain.mRID` (price deltas filtered out):
```diff
3c3
< 	<mRID>82357373cb2549869d47fb207729e635</mRID>
---
> 	<mRID>c2640917bd47447f9a7ca70f628f0e90</mRID>
10c10
< 	<createdDateTime>2022-02-22T20:14:40Z</createdDateTime>
---
> 	<createdDateTime>2022-02-22T20:19:24Z</createdDateTime>
18,19c18,19
< 		<in_Domain.mRID codingScheme="A01">10YNL----------L</in_Domain.mRID>
< 		<out_Domain.mRID codingScheme="A01">10YNL----------L</out_Domain.mRID>
---
> 		<in_Domain.mRID codingScheme="A01">10YBE----------2</in_Domain.mRID>
> 		<out_Domain.mRID codingScheme="A01">10YBE----------2</out_Domain.mRID>
130,131c130,131
< 		<in_Domain.mRID codingScheme="A01">10YNL----------L</in_Domain.mRID>
< 		<out_Domain.mRID codingScheme="A01">10YNL----------L</out_Domain.mRID>
---
> 		<in_Domain.mRID codingScheme="A01">10YBE----------2</in_Domain.mRID>
> 		<out_Domain.mRID codingScheme="A01">10YBE----------2</out_Domain.mRID>
```

**Comment by tvwerkhoven** [comment]
Problem is slightly different: `EntsoePandasClient.query_day_ahead_prices()` does not return duplicate price.amounts. If two (or more?) times have the same price.amount only the first is returned. It turns out the NL 23:00 price is identical to the day before at 02:00.

**Comment by tvwerkhoven** [comment]
I strongly suspect the problem to be this line, commenting it solves it. Not sure why this was added though, seems risky to modify data in a decorator.

https://github.com/EnergieID/entsoe-py/blob/89d53033b70a25784aedfb467e99a08e7fa338f7/entsoe/decorators.py#L101

**Comment by fboerman** [comment]
hi @tvwerkhoven thank you for reporting this. this is indeed a problem with that decorator which was changed in a merged PR which I slipped past my checking.

there should be a duplicated check there but it should only look at the index, that is how it always has been. but the PR changed it to a duplicated call on the dataframe or series itself which means it deletes duplicated prices instead of indexes. a subtle but important difference. I will revert that line and push a new version!
